pytorch_pfn_extras/training/extensions/__init__.py

```diff
@@ -9,20 +9,18 @@
 from pytorch_pfn_extras.training.extensions.log_report import LogReport  # NOQA
 from pytorch_pfn_extras.training.extensions.lr_scheduler import LRScheduler  # NOQA
 from pytorch_pfn_extras.training.extensions.micro_average import MicroAverage  # NOQA
-from pytorch_pfn_extras.training.extensions.print_report import PrintReport  # NOQA
 from pytorch_pfn_extras.training.extensions.profile_report import ProfileReport  # NOQA
-from pytorch_pfn_extras.training.extensions.progress_bar import ProgressBar  # NOQA
 from pytorch_pfn_extras.training.extensions.parameter_statistics import ParameterStatistics  # NOQA
 from pytorch_pfn_extras.training.extensions.plot_report import PlotReport  # NOQA
 from pytorch_pfn_extras.training.extensions.profile_report import ProfileReport  # NOQA
 from pytorch_pfn_extras.training.extensions.value_observation import observe_lr  # NOQA
 from pytorch_pfn_extras.training.extensions.value_observation import observe_value  # NOQA
 from pytorch_pfn_extras.training.extensions.variable_statistics_plot import VariableStatisticsPlot  # NOQA
+from pytorch_pfn_extras.training.extensions import util
 
-try:
-    from pytorch_pfn_extras.training.extensions.print_report_notebook import PrintReportNotebook  # NOQA
-    from pytorch_pfn_extras.training.extensions.progress_bar_notebook import ProgressBarNotebook  # NOQA
-
-    _ipython_module_available = True
-except ImportError:
-    _ipython_module_available = False
+if util._is_notebook():
+    from pytorch_pfn_extras.training.extensions.print_report_notebook import PrintReportNotebook as PrintReport  # NOQA
+    from pytorch_pfn_extras.training.extensions.progress_bar_notebook import ProgressBarNotebook as ProgressBar  # NOQA
+else:
+    from pytorch_pfn_extras.training.extensions.print_report import PrintReport  # type: ignore[misc] # NOQA
+    from pytorch_pfn_extras.training.extensions.progress_bar import ProgressBar  # type: ignore[misc] # NOQA
```

Comment:
This will break existing users relying on `PrintReportNotebook`. Imports for `PrintReportNotebook` `ProgressBarNotebook` should be remained.

Could you also rename `PrintReport` and `ProgressBar` classes to `PrintReportCLI` and `ProgressBarCLI` so that users can explicitly request CLI versions when needed?